components/scream/scripts/gen_boiler.py

```diff
@@ -0,0 +1,1878 @@
+from utils import expect, get_git_toplevel_dir
+
+from collections import OrderedDict
+import pathlib, re, os
+
+#
+# Global hardcoded data
+#
+
+# Templates: maps piece name to generic file text
+FILE_TEMPLATES = {
+    "cxx_bfb_unit_impl": lambda phys, sub, gen_code:
+"""#include "catch2/catch.hpp"
+
+#include "share/scream_types.hpp"
+#include "ekat/ekat_pack.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "physics/{physics}/{physics}_functions.hpp"
+#include "physics/{physics}/{physics}_functions_f90.hpp"
+
+#include "{physics}_unit_tests_common.hpp"
+
+namespace scream {{
+namespace {physics} {{
+namespace unit_test {{
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::{test_data_struct} {{
+
+{gen_code}
+
+}};
+
+}} // namespace unit_test
+}} // namespace {physics}
+}} // namespace scream
+
+namespace {{
+
+TEST_CASE({sub}_bfb, "[{physics}_functions]")
+{{
+  using TestStruct = scream::{physics}::unit_test::UnitWrap::UnitTest<scream::DefaultDevice>::{test_data_struct};
+
+  TestStruct::run_bfb();
+}}
+
+}} // empty namespace
+""".format(physics=phys, sub=sub, test_data_struct=get_data_test_struct_name(sub), gen_code=gen_code),
+
```

Comment:
Personally, I'd prefer to have a row of "#=============" between each of these pieces to make it quicker to tell where one ends and the next starts. These templates are the core of the tool, so we should make them easy to understand. That said, I feel like most people don't like adding lots of #======== lines to their code, so feel free to ignore.

Reply:
Would it make sense to put these templates inside files, like in `scream/scripts/code-templates` or something?

Reply:
I went with @PeterCaldwell 's suggestion of a comment separator to make these more readable. This file has gotten pretty big, so I will keep in mind @bartgol 's thought of perhaps offloading some items to different files.